### DISCUSSION TITLE
deep-learning/attention/transformer-attention

### DISCUSSION THREAD

**Discussion by giscus** [luweizheng/lulaoshi.info#95] [Announcements]
# deep-learning/attention/transformer-attention

Transformer中的注意力机制

https://lulaoshi.info/deep-learning/attention/transformer-attention.html

**Comment by victorming666** [comment]
”第三步：。。。。。且每行总和为0，这是一个权重矩阵。“，这里“每行总和为1”吧？不是0。

**Reply by Hailan-9** [reply]
每行总和是1， 不是0，博主应该是笔误了

**Comment by Hailan-9** [comment]
博主写的太好了，对我刚学习深度学习和transformer的小白来说，真的是深入浅出的入门好资料，结合原论文，基本上理解了原理，点赞点赞

**Comment by QuXiaolong0812** [comment]
great!

**Comment by SumY40** [comment]
看到最细致好懂的一篇

**Comment by xuanyue23** [comment]
写的太好了！读完这个再去读其他文章会清晰很多

**Comment by TureSnow** [comment]
decoder 那里从图中看，第二个 Multi-Head Attention 的 Q 输入是上一个 decoder 输入经过第一个 Multi-Head Attention & mask & Add & Norm 之后的输出，而不是上一个 decoder 的输出？是作者笔误还是图有问题呢？因为从抽象层面看确实是第一个 Multi-Head Attention 经过 mask 和 Add & Norm 之后的输出相对更合理一点，否则第一个多头注意力组件就没有存在的意义了。
